client/src/components/DateButton.tsx

```diff
@@ -0,0 +1,39 @@
+import {Button} from "@material-ui/core";
+import React from "react";
+
+interface DateButtonState {
+    date: string | null;
+}
+// tslint:disable-next-line:no-empty-interface
+interface DateButtonProps {}
+
+export class DateButton extends React.Component<DateButtonProps, DateButtonState> {
+    public state: DateButtonState = {
+        date: null,
+    };
+    constructor(props: DateButtonProps) {
+        super(props);
+    }
+
+    public getDate() {
+        console.log("Date Button works");
+        console.log(
+            fetch("/date")
+                .then( (res) => res.text())
+                .then( (date) => this.setState({date})),
```

Comment:
1. Remove console logs
2. Return the value of fetch, essentially turning this into a function that returns a promise.